tools/jit/gen_jit_dispatch.py

```diff
@@ -195,9 +220,18 @@ def emit_decl(decl):
         all_arguments_are_inputs = tuple(True for _ in arguments)
         only_tensors_are_inputs = tuple(is_tensor_arg(arg) for arg in arguments)
 
+        sized_intlist_args = [i for i, a in enumerate(arguments) if is_sized_intlist_arg(a)]
+        inds_to_replace_powerset = (set(c)
+                                    for r in range(len(sized_intlist_args) + 1)
+                                    for c in combinations(sized_intlist_args, r))
+
         # NB: if there are no scalar args then both options on LHS are equivalent, so deduplicate them.
-        for variant in set([all_arguments_are_inputs, only_tensors_are_inputs]):
-            emit_decl_variant(decl, variant, has_tensorlist)
+        for variant in {all_arguments_are_inputs, only_tensors_are_inputs}:
```

Comment:
Can you please add a comment saying that the set here is important, because those two options might overlap? It's not exactly part of this PR, but you've touched it, and I realized that it's not obvious

Reply:
Hm? You mean the comment above? `# NB: if there are no scalar args then both options on LHS are equivalent, so deduplicate them.`

I didn't touch it

Reply:
Nvm, I'll do that in a followup PR.